### PR TITLE
Add Enclave API

### PR DESCRIPTION
**Pull request by parthsane**

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -33,6 +33,7 @@ const DATA: &'static [(&'static str, &'static [&'static str], &'static [&'static
     ("dxgi1_6", &["basetsd", "dxgi1_2", "dxgi1_4", "dxgi1_5", "dxgitype", "guiddef", "minwindef", "windef", "winnt"], &[]),
     ("dxgiformat", &[], &[]),
     ("dxgitype", &["d3d9types", "dxgiformat", "minwindef"], &[]),
+    ("enclaveapi", &["basetsd", "minwinbase", "minwindef", "ntdef", "winnt"], &[]),
```
Comment (jethrogb) [inline build.rs]: I think you need to add kernel32 in the last array

File: src/um/winnt.rs [hunk 2]
```diff
@@ -4017,6 +4019,7 @@ pub const MEM_IMAGE: DWORD = SEC_IMAGE;
 pub const WRITE_WATCH_FLAG_RESET: DWORD = 0x01;
 pub const MEM_UNMAP_WITH_TRANSIENT_BOOST: DWORD = 0x01;
 pub const ENCLAVE_TYPE_SGX: DWORD = 0x00000001;
+pub const ENCLAVE_TYPE_SGX2: DWORD = 0x00000002;
```
Comment (jethrogb) [inline src/um/winnt.rs:4022]: Is this correct?

Comment (retep998) [reply src/um/winnt.rs:4022]: `#define ENCLAVE_TYPE_SGX2           0x00000002`

**Comment by parthsane** [comment]
@retep998 , is there any thing left which is preventing the PR from going through?

**Comment by retep998** [comment]
Just a matter of me actually getting around to reviewing the PR.

**Comment by jethrogb** [comment]
@retep998 Do you have any idea when you'll have time to do that?

File: src/um/enclaveapi.rs [hunk 3]
```diff
@@ -0,0 +1,66 @@
+// Copyright © 2015-2017 winapi-rs developers
+// Licensed under the Apache License, Version 2.0
+// <LICENSE-APACHE or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// All files in the project carrying such notice may not be copied, modified, or distributed
+// except according to those terms.
+
```
Comment (retep998) [inline src/um/enclaveapi.rs]: Evil blank line.

**Review by retep998 — CHANGES_REQUESTED** [review]

File: src/um/enclaveapi.rs (hunk 3 above)
Comment (jethrogb) [reply src/um/enclaveapi.rs]: :ghost:

**Review by retep998 — APPROVED** [review]

**Comment by parthsane** [comment]
@retep998 Thank you for the merge.
Do you know when we can have a release?

**Comment by jethrogb** [comment]
@retep998 ping

**Comment by retep998** [comment]
I never give ETAs on release because I have no idea when I'll decide to release a new version and how long it'll take me to test for breakages and fix them before releasing. That said, I really should just have an actual schedule for when I do releases.

**Comment by jethrogb** [comment]
Yeah, the last release was almost 6 months ago, I'm sure many people are waiting for features that have since been merged.

**Pull request merged**